pkg/scheduler/preemption/preemption.go

```diff
@@ -341,6 +342,33 @@ type candidateCQ struct {
 	share     int
 }
 
+func cqHeapFromCandidates(candidates []*workload.Info, firstOnly bool, snapshot *cache.Snapshot) *heap.Heap[candidateCQ] {
+	cqHeap := heap.New(
+		func(c *candidateCQ) string {
+			return c.cq.Name
+		},
+		func(c1, c2 *candidateCQ) bool {
+			return c1.share > c2.share
+		},
+	)
+	for _, cand := range candidates {
+		candCQ := cqHeap.GetByKey(cand.ClusterQueue)
+		if candCQ == nil {
+			cq := snapshot.ClusterQueues[cand.ClusterQueue]
+			share, _ := cq.DominantResourceShare()
+			candCQ = &candidateCQ{
+				cq:        cq,
+				share:     share,
+				workloads: []*workload.Info{cand},
+			}
+			_ = cqHeap.PushIfNotPresent(candCQ)
```

Comment:
```suggestion
			cqHeap.Push(candCQ)
```

Since we already checked if the `cqHeap` has `candCQ` in `candCQ == nil`, I'm curious about whether the `Push` function would be enough. WDYT?

Reply:
There is no `Push`, only `PushOrUpdate`. But changed to it, so we can avoid the `_`.

Reply:
Oh, that makes sense. Thank you!